File: taskflow-dotnet-api/src/TaskFlow.Application.Contracts/Users/UserResponseDto.cs
using TaskFlow.Application.Contracts.BaseEntities;
using TaskFlow.Application.Contracts.Roles;

namespace TaskFlow.Application.Contracts.Users;

public class UserResponseDto : AuditedEntityDto<Guid>
{
    public string Email { get; set; } = null!;
    public bool EmailConfirmed { get; set; }
    public string? PhoneNumber { get; set; }
    public bool PhoneNumberConfirmed { get; set; }
    public bool TwoFactorEnabled { get; set; }

    public DateTimeOffset? LockoutEnd { get; set; }
    public bool LockoutEnabled { get; set; }
    public int AccessFailedCount { get; set; }

    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public DateTime? PasswordChangedTime { get; set; }
    public bool IsActive { get; set; }

    public List<RoleResponseDto> Roles { get; set; } = [];
}

File: taskflow-react-client/src/features/users/types/userResponseDto.ts
import type { RoleResponseDto } from "../../roles/types/roleResponseDto";

export interface UserResponseDto {
  id: string;
  email: string;
  emailConfirmed: boolean;
  phoneNumber?: string;
  phoneNumberConfirmed: boolean;
  twoFactorEnabled: boolean;
  lockoutEnd?: string;
  lockoutEnabled: boolean;
  accessFailedCount: number;
  firstName?: string;
  lastName?: string;
  passwordChangedTime?: string;
  isActive: boolean;
  roles: RoleResponseDto[];
  creationTime: string;
  lastModificationTime?: string;
}

